### ISSUE TITLE
Resource save blockchain middleware + pull to refresh

### ISSUE THREAD

**Issue by Nodonisko** [trezor/trezor-suite#10316]
Much more resource savy version of blockchain middleware + pull to refresh functionality to account detail and dashabord.

What to test:
1. Pull to refresh on Dashboard
2. Pull to refresh on Account detail
3. If transaction is received it should appear in Account detail
4. If transaction is confirmed it should change state in Account detail

**Comment by bosomt** [comment]
**QA NOK**

app crashed 4 times of 4 tries
2 times in portfolio tracker and 2 times in account detail

- version 23.12.2 (656) 19c6638 - [develop] - Chore\/receive image btc cash (#10414)
- Android 13 - Nothing phone 1

**Comment by bosomt** [comment]
UPDATE: 

retested one more time

app only crashes when you refresh it during discovery OR during loading of fiat graph.
Please consider disabling this feature during loading of accounts or fiat graph

**Comment by STew790** [comment]
I found something a little different. For me, the app is not crashing while discovery is going on, but when I try to refresh in the account details while the discovery is still going, I see on the device whats in the screenshot below. Note that I tried to refresh a BTC account.
Also, the discovery just won't finish after this.

![image](https://github.com/trezor/trezor-suite/assets/48477147/be085004-c50d-40db-85ae-408b792ecce8)
